src/naturallanguage.cs

```diff
@@ -0,0 +1,246 @@
+
+		[Export ("predictedLabelForString:")]
+		[return: NullAllowed]
+		string GetPredictedLabel (string @string);
+
+		[Export ("predictedLabelsForTokens:")]
+		string[] GetPredictedLabels (string[] tokens);
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface NLTokenizer
+	{
+		[Export ("initWithUnit:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NLTokenUnit unit);
+
+		[Export ("unit")]
+		NLTokenUnit Unit { get; }
+
+		[NullAllowed, Export ("string", ArgumentSemantic.Retain)]
+		string String { get; set; }
+
+		[Internal]
+		[Export ("setLanguage:")]
+		void _SetLanguage (string language);
+
+		[Wrap ("_SetLanguage (language.GetConstant ())")]
+		void SetLanguage (NLLanguage language);
+
+		[Export ("tokenRangeAtIndex:")]
+		NSRange GetTokenRange (nuint characterIndex);
+
+		[Export ("tokensForRange:")]
+		NSValue[] GetTokens (NSRange range);
+
+		[Async (ResultTypeName="NLTokenizerEnumerateContinuation")]
+		[Export ("enumerateTokensInRange:usingBlock:")]
+		void EnumerateTokensInRange (NSRange range, Action<NSRange, NLTokenizerAttributes, bool> block);
+	}
+
+	[iOS (12,0), Mac (10,14, onlyOn64: true), TV (12,0)]
+	[DisableDefaultCtor]
+	[BaseType (typeof(NSObject))]
+	interface NLTagger
+	{
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("initWithTagSchemes:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (NSString[] tagSchemes);
+
+		[Wrap ("this (Array.ConvertAll (tagSchemes, e => e.GetConstant ()))")]
+		IntPtr Constructor (NLTagScheme[] tagSchemes);
+
+		[Internal]
+		[Export ("tagSchemes", ArgumentSemantic.Copy)]
+		NSString[] _TagSchemes { get; }
+
+		[Wrap ("Array.ConvertAll (_TagSchemes, e => NLTagSchemeExtensions.GetValue (e))")]
+		NLTagScheme[] TagSchemes { get; }
+
+		[NullAllowed, Export ("string", ArgumentSemantic.Retain)]
+		string String { get; set; }
+
+		[Static]
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("availableTagSchemesForUnit:language:")]
+		NSString[] GetAvailableTagSchemes (NLTokenUnit unit, NSString language);
+
+		[Static]
+		[Wrap ("Array.ConvertAll (GetAvailableTagSchemes (unit, language.GetConstant()), e => NLTagSchemeExtensions.GetValue (e))")]
+		NLTagScheme[] GetAvailableTagSchemes (NLTokenUnit unit, NLLanguage language);
+
+		[Export ("tokenRangeAtIndex:unit:")]
+		NSRange GetTokenRange (nuint characterIndex, NSString unit);
+
+		[Internal]
+		[NullAllowed, Export ("dominantLanguage")]
+		NSString _DominantLanguage { get; }
+
+		[Wrap ("NLLanguageExtensions.GetValue (_DominantLanguage)")]
+		NLLanguage DominantLanguage { get; }
+
+		[Async (ResultTypeName="NLTaggerEnumerateTagsContinuation")]
+		[EditorBrowsable (EditorBrowsableState.Advanced)]
+		[Export ("enumerateTagsInRange:unit:scheme:options:usingBlock:")]
+		void EnumerateTags (NSRange range, NLTokenUnit unit, NSString scheme, NLTaggerOptions options, Action<NSString, NSRange, bool> block);
```

Comment:
`Action<NSString, NSRange, bool>` is too long (one is good, two if the 2nd is `NSError`)
Please use a custom delegate where the parameter names will act as self-documention

Reply:
Also remove `block` as parameter name, use `handler` or something alike. That said check if that bool isn't as a stop flag (i.e. `out bool`) and if so remove the Async here